### PR TITLE
Add InterceptionEvent

### LINKED ISSUE #222: Add a data model for interceptions
Kloppy's event data model currently does not support interceptions. These are commonly defined as "an attempt by a defending player to intercept a pass between opposition players to prevent the ball from reaching its target.".

### How data providers do it
**StatsBomb**, **Wyscout** and **Opta** all have an "Interception" event. I think **Wyscout** and **Opta** only record successful interceptions (check this; the docs are unclear). **StatsBomb** also records failed interceptions. When an interception is immediately followed by a pass, **StatsBomb** and **Wyscout** do not record an "Interception" event and the pass gets typed/tagged as an “Interception”. Maybe, these passes have to be split into two events again when doing the conversion to the Kloppy format.

### Proposal
Add an `InterceptionEvent` with an `InterceptionResult`. As possible values for the result, I would suggest:
- Success: An interception that gains possession of the ball (without going out of bounds)
- Lost: An interception by the defending team that knocked the ball to an attacker
- Out: An interception that knocked the ball out of bounds

Maybe for Wyscout and Opta, the result can be inferred from the next action.

Passes that are labeled as an interception should by split into a "Interception" and "Pass" event.

### PR DESCRIPTION
**Pull request by MKlaasman**
fixes #222 

Notes:
- Whenever needed (wyscout & opta), I inferred the InterceptionResult from the next event. Though, this approach might still be suboptimal.
- In WyScout v2: The orginal RecoveryEvent gets overwritten by the InterceptionEvent, whenever possible

What do you think? @probberechts  @JanVanHaaren

### CODE REVIEW

File: kloppy/infra/serializers/event/opta/deserializer.py [hunk 1]
```diff
@@ -761,6 +792,14 @@ def deserialize(self, inputs: OptaInputs) -> EventDataset:
                             **duel_event_kwargs,
                             **generic_event_kwargs,
                         )
+                    elif type_id == EVENT_TYPE_INTERCEPTION:
```
Comment (probberechts) [inline kloppy/infra/serializers/event/opta/deserializer.py:795]: Opta also has a "blocked pass" event. Would it make sense to parse it as an interception too?

blocked pass := When a player tries to cut out an opposition pass by any means. Similar to an interception except there is much less reading of the pass.

File: kloppy/infra/serializers/event/wyscout/deserializer_v2.py [hunk 2]
```diff
@@ -477,6 +510,24 @@ def deserialize(self, inputs: WyscoutInputs) -> EventDataset:
                         **generic_event_args,
                     )
 
+                # If also an interception event, add this before other event (It is a tag on multiple wyscout_events)
```
Comment (probberechts) [inline kloppy/infra/serializers/event/wyscout/deserializer_v2.py:513]: Wyscout does not have a separate event type for interceptions. Instead, interceptions are recorded as a pass, duel or touch event with an "interception" tag. Your current implementation works well for passes, but for consistency with other providers (and socceraction) I would convert these duels/touch events to an interception. Not to an interception + duel.

**Review by probberechts — COMMENTED** [review]
I've only added two suggestions. Apart from these it seems ok to me. Thanks for implementing this!

Note that I did not check the Wyscout v3 implementation in detail since I am not familiar with that data.

**Comment by MKlaasman** [comment]
Addition:
1) Wyscout v2: Now duels and touches are overwritten when they have an interception tag. 
2) Opta: Handle a blocked pass as an interception.
3) Opta: Filtered deleted events upfront, otherwise next_event implementation prone for wrong InterceptionResult. Now, not logging the Skipping of events anymore. Someone, thinking this is a bad design choice?

**Review by probberechts — APPROVED** [review]

**Review by JanVanHaaren — APPROVED** [review]
This pull request looks good to me. Thanks!

**Comment by koenvo** [comment]
@MKlaasman thanks for your work. Can you please merge master in? There are some small merge conflicts due to changes from Pieters PRs.

**Pull request merged**